### PR TITLE
Channel specific hubot commands

### PR DESCRIPTION
**Pull request by kristenmills**
### Overview of how this works
The primary officers will be set to admins. Admins are the only people who have the ability to assign roles. Roles can be assigned only in your private chats with hubot because they don't have ids associated with the commands.  

* `hubot <user> doesn't have <role> role` - Removes a role from a user
* `hubot <user> has <role> role` - Assigns a role to a user

There are two roles that can allow and remove commands: admin and '[channelname]-admin' . If you possess one of those roles, you may run the following commands: 
* `hubot enable <commandId>`
* `hubot disable <commandId>`
* `hubot enable all`
* `hubot disable all`

and they will apply to the specific channel your in.  Your DM with hubot allows for all commands.

To see all commmands you can run
```hubot list commands```

They are organized into disabled and enabled for a channel. It's sent via DM to you to prevent spam.

### Todo List

* ~~`hubot enable/disable all` - enables or disables all command. Does not disable commands that can't be disabled (help, enable/disable, list commands)~~ Done!

This is not ready to be deployed but it's a stable point where I want feedback.

### CODE REVIEW

**Comment by weswigham** [comment]
It seems okay; but why did `scoreboard.coffee` get removed with this?

File: scripts/room.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+//   hubot allow command <command id> - Allows this command in the current room
+//   hubot remove command <command id > - Remove this command in the current room
+
+module.exports = function(robot) {
+
+  robot.respond(/allow command (.*)/i, {id: 'room.allow'}, function(msg) {
+    var plugin = msg.match[1].toLowerCase();
+    var room = msg.message.room;
+
+    var roomPermissions = robot.brain.data.roomPermissions = robot.brain.data.roomPermissions || {};
```
Comment (benkantor) [inline scripts/room.js]: Is there a reason this isn't just 

    var roomPermissions = robot.brain.data.roomPermissions || {};

Comment (weswigham) [inline scripts/room.js]: This sets `robot.brain.data.roomPermissions` to `{}` if it doesn't exist.

Comment (benkantor) [inline scripts/room.js]: Oh, that makes sense, nevermind

**Comment by kristenmills** [comment]
@weswigham  I removed `scoreboard.coffee` because we are going to have that functionality in [slushy](https://github.com/rit-sse/slushy).

**Comment by kristenmills** [comment]
The help thing looks like too much of an annoyance so I'm going to nix that.

**Comment by kristenmills** [comment]
You know what would make more sense: A blacklist approach over a whitelist approach.
Imma do that.

**Comment by weswigham** [comment]
Conditional black lists, so you can ban 'pug bomb 20' but not 'pug bomb 3'?

**Comment by kristenmills** [comment]
@weswigham Yeah make a separate issue pls. I wanna get the basics done and merged before hand.

**Comment by kristenmills** [comment]
I'm probably going to merge this by the end of the day unless anyone complains.

**Comment by kristenmills** [comment]
Alright, it looks like middleware features being merged into hubot might happen in the foreseeable future. (see github/hubot#803). I'm going to wait that out before merging this branch.

**Comment by michaelansel** [comment]
This is awesome! Once #803 merges, you should pull the command blacklist stuff out to a separate module or at least write up a bit to go in [patterns.md](https://github.com/github/hubot/blob/master/docs/patterns.md).

File: scripts/command-middleware.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+//
+module.exports = function(robot) {
+
+  robot.listenerMiddleware(function(context, next, done){
+    var id = context.listener.options.id;
+    var room = context.response.envelope.room;
+    var user = context.response.envelope.user;
+    var cb = robot.brain.data.commandBlacklists || {};
+    var blacklist = cb[room] || [];
+    if (room === user.name ) { //I don't care what you and hubot do on your own time
```
Comment (craigcabrey) [inline scripts/command-middleware.js]: What happens in `user.name` stays in `user.name`.

**Comment by kristenmills** [comment]
@michaelansel Thanks! Yeah I can definitely do that.

**Pull request merged**